lib/streamlit/runtime/scriptrunner/script_run_context.py

```diff
@@ -79,6 +79,9 @@ class ScriptRunContext:
     fragment_ids_this_run: set[str] | None = None
     # we allow only one dialog to be open at the same time
     has_dialog_opened: bool = False
+    # If true, it indicates that we are in a cached function that disallows
+    # the usage of widgets.
+    disallow_cached_widget_usage: bool = False
```

Comment:
nit: I find doubled negations like `disallow = not allow_widgets` as used [above](https://github.com/streamlit/streamlit/pull/8503/files#diff-e0869bb97596a8ea14c2972a0815c4b3e762f9e8fb04d7a5715790ab7cef5b5eR263) can be harder to read vs `allow = allow_widgets`
So maybe worth considering renaming it, but no strong opinion

Reply:
if it was `allow_cached_widget_usage`, what would be the default value?

Reply:
Had an offline conversation about this with Lukas and he is going to update the comment to explain the reason for this naming

Reply:
The direct negation within the code logic doesn't exist anymore. Not sure where to put a comment since I think using `disallow_cached_widget_usage` as a flag also seems fine in general.